src/recall-assistant/ift-service.ts

```diff
@@ -156,48 +180,39 @@ export async function getTransactions(req, inputEpcs: string[]) {
          2019-11-15&epc_id[]=urn%3Aibm%3Aift%3Aproduct%3Alot%3Aclass%3A1953084565871.APJj.2322&epc_id[]=urn%3Aibm%3
          Aift%3Aproduct%3Alot%3Aclass%3A1953084565871.pFOa.2131212&epc_id[]=urn%3Aibm%3Aift%3Aproduct%3Alot%3Aclass
          %3A1953084565871.OdCD.475" */
-
-      options.uri = eventCallUri;
-      // Issue request to the trace API and save in an array of promises
-      eventsPromiseList.push(rp(options));
-    }
+      return eventCallUri;
+    }));
   }
 
-  const transactionIds = await processPromiseList(eventsPromiseList, 'aggregation');
+  const transactionIds = processPromiseList(eventsList, 'aggregation');
   return transactionIds;
 }
 
-export async function processPromiseList(promiseList: string[], eventType: string) {
-  return Promise.all(promiseList).then((eventResponse: any) => {
-    let ids: string[] = []; // Get a list of all EPCs listed as outputs on these tranformations or transactionIds
-    eventResponse.forEach((response) => {
-      const eventsObj = JSON.parse(response);
-      eventsObj.events.forEach((event) => {
-        if (eventType === 'aggregation') {
-          // Loop through transactions on each event to get ids
-          event.transaction_ids.forEach((transaction) => {
-            ids = [...ids, transaction.id];
-          });
-        } else if (eventType === 'transformation') {
-          event.output_quantities.forEach((output) => {
-            ids = [...ids, output.epc_id];
-          });
-        }
-      });
+export function processPromiseList(responseList: any[], eventType: string) {
```

Comment:
Maybe this now needs to be `processResponse` instead? Since the promises are already processed earlier and we only get the responses as input to this method?

Reply:
Yes, that would be good. I meant to rename it to `processResponse` or `processResponseList` but I must have forgot.